Scripts/RemediationScripts/Remediate-EncryptAutomationAccountVariables

```diff
@@ -0,0 +1,684 @@
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    
+    # Setting up context for the current Subscription.
+    $context = Set-AzContext -SubscriptionId $SubscriptionId -ErrorAction Stop
+    
+
+    if(-not($AutoRemediation))	
+    {
+        Write-Host "Subscription Name: [$($context.Subscription.Name)]"
+        Write-Host "Subscription ID: [$($context.Subscription.SubscriptionId)]"
+        Write-Host "Account Name: [$($context.Account.Id)]"
+        Write-Host "Account Type: [$($context.Account.Type)]"
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "To encrypt the unencrypted variables associated with the Automation Accounts in a Subscription, Contributor or higher privileges on the Automation Accounts are required." -ForegroundColor $([Constants]::MessageType.Warning)
+    Write-Host $([Constants]::SingleDashLine)
+
+    Write-Host "[Step 2 of 5] Fetch all Automation Accounts"
+    Write-Host $([Constants]::SingleDashLine)
+    $automationAccountResources = @()
+    $requiredMinTLSVersion = 1.2
+
+    # To keep track of remediated and skipped resources
+    $logRemediatedResources = @()
+    $logSkippedResources=@()
+    
+    # Control Id
+    $controlIds = "Azure_AutomationAccounts_DP_Encrypt_Variables"
+
+    if($AutoRemediation)
+    {
+        if(-not (Test-Path -Path $Path))
+        {
+            Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::DoubleDashLine)
+            return
+        }
+        Write-Host "Fetching all Automation Accounts failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+        $controls = $controlForRemediation.ControlRemediationList
+        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}
+
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            Write-Host "No Automation Account found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
+            Write-Host $([Constants]::DoubleDashLine)
+            return
+        } 
+
+        $allAutomationAccountResources = Get-AzAutomationAccount -ErrorAction SilentlyContinue
+        $validResources | ForEach-Object { 
+            try
+            {
+                $name = $_.ResourceName
+                $resourceGroupName = $_.ResourceGroupName
+                $automationAccountResources += $allAutomationAccountResources | Where-Object {$_.AutomationAccountName -eq $name -and $_.ResourceGroupName -eq $resourceGroupName}
```

Comment:
If resource was there in file, but not present anymore in subscription. We should log that in skipped resource wth reason as resource not found.

Reply:
Addressed